### PR TITLE
[1/2] feat(gemini): add text Interactions Responses conversion

### PR DESCRIPTION
**Pull request by IceCodeNew**
# [1/2] feat(gemini): add text Interactions Responses conversion

## Description

Add non-streaming Gemini text Interactions through the Responses API, including thought-token usage and failure diagnostics. Require `google-genai>=2.17.0` for `Interaction.errors`; Vertex's SDK floor and capabilities remain unchanged.

## Review focus

- In `providers/gemini`, check the text-only boundary against Google's [Interactions API](https://ai.google.dev/gemini-api/docs/interactions/interactions-overview). Unsupported tools, media, chaining and `store` are rejected. Rejecting `store` leaves Google's default server-side storage in effect.
- Check per-operation `v1` selection, without changing generateContent defaults. Thought/signature-only output is omitted; thought tokens count toward generated usage.

## Verification

- Compared real-SDK HTTP conversion with Google's [v2.17.0 Interaction model](https://github.com/googleapis/python-genai/blob/v2.17.0/google/genai/_gaos/types/interactions/interaction.py) and [v2.23.0 source](https://github.com/googleapis/python-genai/tree/v2.23.0). The v2.16.0 predecessor fails on missing `errors`; v2.17.0 supplies the required field. Request, usage and structured-output checks cover the projection boundaries.
- Fresh-context reviewers checked full aggregate diffs and tests, prioritizing baseline compatibility, official sources, then Eino/Fantasy. Behavior probes and ablations preceded repairs and a rebuilt aggregate review.
- Ran offline unit, lint and typing checks. Independent runs retained baseline typing errors and unavailable Watsonx dependencies; live account, region and model acceptance remain unverified.

<details>
<summary>Skills and sources</summary>

- [`karpathy-guidelines`](https://github.com/forrestchang/andrej-karpathy-skills/blob/2c606141936f1eeef17fa3043a72095b4765b9c2/skills/karpathy-guidelines/SKILL.md)
- [`reclaim-code-entropy`](https://github.com/Yevanchen/reclaim-code-entropy/blob/9e6482e45814076b9710b55c8dcbb064ba4b7977/skills/reclaim-code-entropy/SKILL.md)
- [`scoped-change`](https://github.com/scarletkc/agents/blob/d5a312346adcf2169eb5a31a1b88368441b32327/skills/scoped-change/SKILL.md)
- [`ux-writing`](https://github.com/scarletkc/agents/blob/d5a312346adcf2169eb5a31a1b88368441b32327/skills/ux-writing/SKILL.md)
- [`thermos`](https://github.com/cursor/plugins/blob/bdf7aa355337897f167153e05069aca505dae17c/thermos/skills/thermos/SKILL.md)
- [`friendly-python`](https://github.com/PsiACE/skills/blob/2265aed05caf199426a8062461e2c9901be996d8/skills/friendly-python/SKILL.md)
- [`piglet`](https://github.com/PsiACE/skills/blob/2265aed05caf199426a8062461e2c9901be996d8/skills/piglet/SKILL.md)
- [`japanese-tech-writing`](https://gist.githubusercontent.com/k16shikano/fd287c3133457c4fd8f5601d34aa817d/raw/8f2d57610a73efc97d743c9b0b0ecb1002e09fa4/SKILL.md), applied to English prose
- `better-goal`: local planning adaptation of [OpenAI goal guidance](https://developers.openai.com/cookbook/examples/codex/using_goals_in_codex)

</details>

## PR Type

New feature.

## Checklist

- [ ] I understand the code I am submitting. (Human author confirmation pending.)
- [x] I have added unit tests that prove my fix/feature works
- [x] I have run this code locally and verified it fixes the issue. (Offline tests.)
- [x] New and existing applicable tests pass locally; independent environment limits are disclosed above
- [x] Documentation was updated where necessary
- [x] I have read and followed the [contribution guidelines](https://github.com/mozilla-ai/any-llm/blob/main/CONTRIBUTING.md)
- [x] **AI Usage:**
    - [ ] No AI was used.
    - [x] AI was used for drafting/refactoring.
    - [ ] This is fully AI-generated.

## AI Usage Information

- AI Model used: GPT-6 Astra Medium
- AI Developer Tool used: Amp
- [x] I am an AI Agent filling out this form (check box if true)

## Summary by CodeRabbit

- **New Features**
  - Added support for Gemini Interactions through the Responses API for text-only requests.
  - Gemini interaction results are converted into the standard Responses format, including status, usage, and timestamp details.
  - Unsupported parameters, input types, and output content now return clear errors.
  - Existing completion requests continue to use Gemini’s standard content-generation API.
  - Additional request headers, query parameters, and body options are forwarded correctly.

- **Documentation**
  - Added Gemini Interactions guidance to the quickstart guide, including supported options, storage defaults, and timeout behaviour.

### CODE REVIEW

File: src/any_llm/providers/gemini/interactions.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+) -> OpenAIResponse:
+    status = _map_status(interaction.status)
+    previous_response_id = None
+    instructions = None
+    metadata = None
+    response_error = None
+    if not isinstance(interaction, InteractionSseEventInteraction):
+        previous_response_id = interaction.previous_interaction_id
+        instructions = interaction.system_instruction
+        metadata = interaction.labels
```
[the remark this answers is not in the document]
Comment (IceCodeNew) [reply src/any_llm/providers/gemini/interactions.py:145]: @coderabbitai, thanks for flagging this. These paths remain as prerequisites for #1383, which descends from this PR. Its public Responses HTTP/SSE tests exercise stream parameter forwarding and terminal InteractionSseEventInteraction conversion through the real SDK.

The dependent implementation and transport tests are now published in https://github.com/mozilla-ai/any-llm/commit/43ffc283c92cedf6decaa32fe0abf7e07d6fd8d5. The owner suites and independent minimum/latest SDK probes passed; removing these paths would break that streaming consumer.

**Comment by IceCodeNew** [comment]
@coderabbitai review

**Comment by IceCodeNew** [comment]
@coderabbitai full review

**Comment by IceCodeNew** [comment]
IceCodeNew added absent and empty timestamp coverage while retaining the invalid-value and unknown-status checks in https://github.com/IceCodeNew/any-llm/commit/0ba1e46. The Gemini interaction suite and all pre-commit hooks, including mypy, pass. The same commit is included in #1383.

**Comment by IceCodeNew** [comment]
@coderabbitai resume

**Comment by IceCodeNew** [comment]
Fixed the historical outside-diff transport-options finding: aresponses now separates extra_headers and extra_query before strict parameter validation. The public real-SDK transport test reproduced the rejection before the fix and now checks the actual header and query string. Full unit tests and pre-commit pass after rebasing onto upstream main.

**Comment by IceCodeNew** [comment]
Fixed the missing-model fallback after a public real-SDK HTTP test reproduced an empty Response.model. The test also verifies that a returned model name takes precedence. Clarified store=True/store=False rejection in the quickstart and corrected the ordinary usage fixture; explicit-total preservation controls remain unchanged. Full units and pre-commit pass.

@coderabbitai Please reassess the copyright-header finding: pinned Ruff 0.15.20 reports CPY001 as preview-only, and this repository does not enable preview. The actual configured nonfixing check on the new files passes, as does pre-commit run --all-files. Selecting ALL does not enable preview rules by itself. No copyright or lint configuration change is warranted by this finding.

**Comment by njbrake** [comment]
Sorry for the silence. You opened #1337 on August 24 and iterated on #1338 for two weeks without a response from us. That is on me, not you, and it left you making an architectural call alone that we should have answered up front. Thank you for the work regardless, and particularly for testing the conversion against the real SDK over a mock transport rather than mocking the client. That caught things a client mock would have hidden.

**The call, now made:** mapping `responses()` onto Interactions is right for any-llm. Google publishes no OpenAI-style `/responses` endpoint, so there is no vendor surface to defer to, and Interactions is their primary API going forward. A future `interactions()` method can carry the stateful surface (chaining, typed steps, agents) the way `messages()` carries Anthropic's, so the text subset here is a façade rather than a dead end.

_Note: this comment was drafted by Claude Opus 5 via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

**Review by njbrake — APPROVED** [review]
Approving the implementation. Text Interactions conversion, the transport-option fix in `any_llm.py`, and the test coverage all hold up.

Verified live: `test_responses_async[gemini]` passes against the real Interactions endpoint after dropping the pinned `api_version="v1"`. Structured-output tests skip on `UnsupportedParameterError`, which is expected for this text-only slice.

Remaining, not blocking: reference the issue as `Refs #1337` rather than `Fixes`, since `get`, chaining, typed steps, tools and background stay open. #1383 needs a rebase onto this.

_Note: this review was drafted by Claude Opus 5 via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._
